sdk/src/sdk/vdb/grpc/kuksa_val_v2/BrokerClient.cpp

```diff
@@ -233,12 +235,15 @@ class MetadataRequester {
     }
 
     void onMetadataResponse(const kuksa::val::v2::ListMetadataResponse& response) {
-        assert(response.metadata_size() == 1);
-        if (response.metadata_size() != 0) {
+        if (response.metadata_size() == 1) {
             m_metadataStore->addMetadata(std::make_shared<Metadata>(
                 Metadata{*m_currentPath, response.metadata(0).id(), true}));
             getNextMetadata();
         } else {
+            logger().error("Requesting metadata of signal '{}' results in metadata for {} signals "
+                           "returned. Assuming the signal as 'unknown'.",
+                           *m_currentPath, response.metadata_size());
+            assert(response.metadata_size() == 1);
```

Comment:
isn't that assert now always triggered? size in the else branch should always be != 1

Reply:
assert removed - error case is sufficiently handled by the else case